### PR TITLE
atmega_common/gpio.c Fixes GPIO_LOW interrupt

### PR DESCRIPTION
**Pull request by shr70**
fixes the GPIO_LOW interrupt on the atmega platform.

### Contribution description
This PR fixes a bug in the atmega_common gpio.c file. In order to set a GPIO_LOW interrupt, EICRA will need to set some bits to zero. The originial code did this by just shifting the `flank` variable. But 0 variables can not be shifted. Therefore the interrupts were not set correctly. Its fixed by checking first if GPIO_LOW is supposed to be set and then using `EICRA &= ~(0x3 << (int_num * 2));` to set the GPIO_LOW interrupt correctly.

### CODE REVIEW

File: cpu/atmega_common/periph/gpio.c [hunk 1]
```diff
@@ -174,11 +174,21 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
 
     /* apply flank to interrupt number int_num */
     if (int_num < 4) {
-        EICRA |= (flank << (int_num * 2));
+        if (flank == GPIO_LOW) {
```
Comment (ZetaR60) [inline cpu/atmega_common/periph/gpio.c]: This issue also exists with all flanks that contain zeros (i.e. anything but GPIO_RISING).

I would suggest removing the check and instead always zero the relevant bits, followed by writing the flank onto them.

File: cpu/atmega_common/periph/gpio.c [hunk 2]
```diff
@@ -174,11 +174,21 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
...
+            EICRA &= ~(0x3 << (int_num * 2));
+        }
+        else {
+            EICRA |= (flank << (int_num * 2));
+        }
     }
 #if defined(EICRB)
     else {
-        EICRB |= (flank << ((int_num % 4) * 2));
+        if (flank == GPIO_LOW) {
```
Comment (ZetaR60) [inline cpu/atmega_common/periph/gpio.c]: See above.

File: cpu/atmega_common/periph/gpio.c [hunk 3]
```diff
@@ -244,54 +254,46 @@ static inline void irq_handler(uint8_t int_num)
     __exit_isr();
 }
 
-ISR(INT0_vect, ISR_BLOCK)
-{
+ISR(INT0_vect, ISR_BLOCK){
```
Comment (ZetaR60) [inline cpu/atmega_common/periph/gpio.c]: Why did you change the style of unrelated code? Several other instances below.

Comment (shr70) [reply cpu/atmega_common/periph/gpio.c]: I missed this change. I ran uncrustify before pushing the branch. It does not seem to recognize ISR as a function and moved the curly brace therefore. I changed it back now.

Comment (ZetaR60) [reply cpu/atmega_common/periph/gpio.c]: Looks like there is a fix being worked on for this issue: #9370

**Comment by ZetaR60** [comment]
Unrelated failure in Murdock, but other than that it looks good. Please squash your fix into the first commit (git rebase then force push). For future use: Git provides a convenient way to label commits as fixes for automatic squashing (using ```--fixup``` for git commit).
